WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java

```diff
@@ -86,6 +86,12 @@ public void onPause() {
     @Override
     public void onResume() {
         super.onResume();
+
+        // Site name may have changed (e.g. via Settings and returning to this Fragment) so update the UI
+        if (mBlog != null && !mBlogTitleTextView.getText().equals(mBlog.getBlogName())) {
```

Comment:
Hmm, indeed. I used this method: https://github.com/wordpress-mobile/WordPress-Android/blob/issue/3607-site-name-refresh/WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java#L370-L372

Happy to change it to the one you've highlight though if that's more robust! I'm sure i'm missing some background on this one.